.circleci/config.yml

```diff
@@ -446,6 +447,55 @@ jobs:
           condition: *deploy-condition
           steps:
             - gcp-gcr/push-image: *private-image
+  generate-diff:
+    docker: *docker
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - *restore_venv_cache
+      - *build
+      - *restore_mvn_cache
+      - *java_deps
+      - run:
+          name: Generate SQL content diff
+          command: |
+            # compare a branch against the main branch,
+            # or skip if we're already on the main branch
+            if [[ "$CIRCLE_BRANCH" == "main" ]]; then
+              circleci-agent step halt
+            fi
+            git clone --single-branch --branch main \
+              git@github.com:mozilla/bigquery-etl \
+              bigquery-etl-main
+
+            cd bigquery-etl-main
+            pip install -r requirements.txt
+            pip install -r java-requirements.txt
+            mvn dependency:copy-dependencies
+            ./script/generate_sql \
+              --target-project moz-fx-data-shared-prod
```

Comment:
I guess we could also just compare to the `generated-sql` branch here. Pulling in the main branch, however, ensures that we actually compare it to the most recent data (e.g. when pushing to `main` there is a delay once generated SQL gets pushed to `generated-sql`, all PRs opened during that time would compare SQL to an old state). The SQL generation is relatively fast, so it might not make a huge difference here.

Reply:
I think this is fine. We can watch timings and optimize in the future if this becomes a problem.